brian2/equations/equations.py

```diff
@@ -433,6 +437,25 @@ def __init__(self, type, varname, dimensions, var_type=FLOAT, expr=None,
                                                       if variable =='xi' or variable.startswith('xi_')]),
                                     doc='Stochastic variables in the RHS of this equation')
 
+
+    _state_tuple = property(lambda self: (self.type, self.varname,
+                                          self.dim, self.var_type,
+                                          self.expr, tuple(self.flags)),
+                            doc='A tuple representing the full state of this '
+                                'object, used for hashing and equality '
+                                'testing.')
```

Comment:
I see your point, but any change to those core classes has to be done very carefully -- there are a lot of other possibilities to break this apart from the `_state_tuple`. I don't see a straightforward way to make this specific thing safer with regard to future changes, IMO having extensive tests is the most important...